src/content/en/updates/posts/2015/11/app-shell.markdown

```diff
@@ -113,13 +122,13 @@ During the first load experience, your goal is to get meaningful content to the
 
 * Lazy-load or background load everything else. One good option is to use [read-through caching](https://googlechrome.github.io/samples/service-worker/read-through-caching/) for dynamic content.
 
-* Use service worker tools ([sw-precache](https://github.com/GoogleChrome/sw-precache)) for reliable caching and updating of service worker to automatically manage the caching of static content.
+* Use service worker tools, such as [sw-precache](https://github.com/GoogleChrome/sw-precache), for example to reliably cache and update the service worker that manages your static content. (More about sw-precache later.)
 
 **To achieve this:**
 
-* **Server** will send down HTML content the client can render and will use far-future HTTP cache expiration headers to account for browsers without service worker support. It will serve filenames using hashes to enable ‘versioning’ and easy updates for later in the application lifecycle.
+* **Server** will send HTML content that the client can render and use far-future HTTP cache expiration headers to account for browsers without service worker support. It will serve filenames using hashes to enable both ‘versioning’ and easy updates for later in the application lifecycle.
 
-* **Page(s)** will include inline CSS styles in a `<style>` tag within the document `<head>` to provide a fast first paint of the application shell. Each page will asynchronously load the JavaScript necessary for the current view. As we all know, CSS cannot yet be asynchronously loaded.. Instead, we can request styles using JavaScript as it IS asynchronous rather than parser-driven and synchronous. We can also take advantage of requestAnimationFrame to avoid cases where we might get a fast cache hit and end up with styles accidentally becoming part of the critical rendering path. requestAnimationFrame forces the first frame to be painted and then the styles to get loaded. Another option is to use projects such as Filament Group’s [loadCSS](https://github.com/filamentgroup/loadCSS) to request CSS async using JavaScript.
+* **Page(s)** will include inline CSS styles in a `<style>` tag within the document `<head>` to provide a fast first paint of the application shell. Each page will asynchronously load the JavaScript necessary for the current view. As we all know, CSS cannot yet be asynchronously loaded. Instead, we can request styles using JavaScript as it IS asynchronous rather than parser-driven and synchronous. We can also take advantage of [`requestAnimationFrame()`](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) to avoid cases where we might get a fast cache hit and end up with styles accidentally becoming part of the critical rendering path. `requestAnimationFrame()` forces the first frame to be painted before the styles to are loaded. Another option is to use projects such as Filament Group’s [loadCSS](https://github.com/filamentgroup/loadCSS) to request CSS asynchronously using JavaScript.
```

Comment:
"painted before the styles to are loaded" -> "painted before the styles to be loaded"